heron/common/src/java/org/apache/heron/common/basics/ResourceMeasure.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.heron.common.basics;
+
+public interface ResourceMeasure<T> extends Comparable<T> {
+
+  boolean isZero();
+
+  T minus(T other);
+
+  T plus(T other);
+
+  T multiply(int factor);
+
+  T divide(int factor);
+
+  T increaseBy(int percentage);
+
+  boolean greaterThan(T other);
```

Comment:
The `Comparable` interface already provides the `compareTo` method. Do we really need these 4 comparison methods?